src/main/kotlin/uk/gov/justice/digital/hmpps/educationandworkplanapi/app/client/prisonapi/PrisonApiMapper.kt

```diff
@@ -0,0 +1,71 @@
+package uk.gov.justice.digital.hmpps.educationandworkplanapi.app.client.prisonapi
+
+import org.springframework.stereotype.Component
+import uk.gov.justice.digital.hmpps.prisonapi.resource.model.PrisonerInPrisonSummary
+import uk.gov.justice.digital.hmpps.prisonapi.resource.model.SignificantMovement
+import uk.gov.justice.digital.hmpps.prisonapi.resource.model.TransferDetail
+
+@Component
+class PrisonApiMapper {
+  fun toPrisonMovementEvents(
+    prisonerNumber: String,
+    prisonerInPrisonSummary: PrisonerInPrisonSummary,
+  ): PrisonMovementEvents {
+    val prisonMovements = mutableMapOf<Long, List<PrisonMovementEvent>>()
+
+    prisonerInPrisonSummary.prisonPeriod?.forEach {
+      val movementEvents = mutableListOf<PrisonMovementEvent>()
+      movementEvents.addAll(buildArrivalsAndReleases(it.movementDates))
+      movementEvents.addAll(buildTransfers(it.transfers))
+      prisonMovements[it.bookingId!!] = movementEvents
+    }
+
+    return PrisonMovementEvents(prisonerNumber, prisonMovements)
+  }
+
+  private fun buildArrivalsAndReleases(movements: List<SignificantMovement>): List<PrisonMovementEvent> {
+    val arrivalsAndReleases = mutableListOf<PrisonMovementEvent>()
+    movements.forEach {
+      if (isAdmissionIntoPrison(it)) {
+        arrivalsAndReleases.add(toPrisonArrivalEvent(it))
```

Comment:
Minor, but we might want to check with someone like Andy Marke what the official terminology is re: admissions.
They appear to have called it admissions (ADM), but we are using the term arrival ... I wonder if there's any mileage in being more aligned in this respect and it's potentially some key HMPPS terminology ?

Reply:
I'll use `Arrivals` instead, as that's what they've called it (at least most of the time - I've found `Arrivals` in their spec too!).